File: Practice/10/python/10.py
s, l1, r1, l2, r2 =  input('Введите числа s, l₁, r₁, l₂, r₂:\n').split()
x = int(l1)
y = int(r2)

if x + y < int(s):
  x = min(int(s)-y, int(r1))
else:
  y = max(int(s)-x, int(l2))

if x + y == int(s):
  print(x, y)
else:
  print(-1)

File: Practice/10/c++/10/10.cpp

#include <iostream>
using namespace std;

int main()
{
	setlocale(LC_ALL, "RUSSIAN");
	int s, l1, r1, l2, r2;
	cout << "Введите числа s, l1, r1, l2, r2:\n";
	cin >> s >> l1 >> r1 >> l2 >> r2;
	int x = l1;
	int y = r2;

	if (x + y < s) {
		x = min(s - y, r1);
	}
	else {
		y = max(s - x, l2);
	}

	if (x + y == s) {
		cout << x << " " << y;
	}
	else {
		cout << -1;
	}
}